enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+> enforced on my cluster."
+
+> "As a cluster lifecycle administrator, I want to be able to diagnose why pending changes have not been applied
+> if I expected them to be."
+
+> "As a cluster administrator or privileged user familiar with OpenShift prior to the introduction of change management, 
+> I want it to be clear when I am looking at the desired versus actual state of the system. For example, if I can see 
+> the state of the clusterversion or a machineconfigpool, it should be straightforward to understand why I am 
+> observing differences in the state of those resources compared to the state of the system."
+
+### Goals
+
+1. Indirectly support the strategic separation of control-plane and worker-node update phases for Standalone clusters by supplying a change control mechanism that will allow both control-plane and worker-node updates to proceed at predictable times without doubling operational overhead.
+2. Directly support the strategic separation of control-plane and worker-node update phases by implementing a "manual" change management strategy where users who value the full control of the separation can manually actuate changes to them independently.
+3. Empower OpenShift cluster lifecycle administrators with tools that simplify implementing industry standard notions of maintenance windows.
+4. Provide Service Delivery a platform native feature which will reduce the amount of custom tooling necessary to provide maintenance windows for customers.
+5. Deliver a consistent change management experience across all platforms and profiles (e.g. Standalone, ROSA, HCP).
+6. Enable SRE to, when appropriate, make configuration changes on a customer cluster and have that change actually take effect only when permitted by the customer's change management preferences.
+7. Do not subvert expectations of customers well served by the existing fully self-managed cluster update.
+8. Ensure the architectural space for enabling different change management strategies in the future. 
+
+### Non-Goals
+
+1. Allowing control-plane upgrades to be paused midway through an update. Control-plane updates are relatively rapid and pausing will introduce unnecessary complexity and risk. 
+2. Requiring the use of maintenance schedules for OpenShift upgrades (the changes should be compatible with various upgrade methodologies – including being manually triggered).
+3. Allowing Standalone worker-nodes to upgrade to a different payload version than the control-plane (this is supported in HCP, but is not a goal for standalone).
+4. Exposing maintenance schedule controls from the oc CLI. This may be a future goal but is not required by this enhancement.
+5. Providing strict promises around the exact timing of upgrade processes. Maintenance schedules will be honored to a reasonable extent (e.g. upgrade actions will only be initiated during a window), but long-running operations may exceed the configured end of a maintenance schedule.
```

Comment:
>How far into a maintenance schedule's window will an update event trigger?

Change can be initiated up to the last instant of the permissive window. 

> if something stops the upgrade of workers to start, maybe the operator doing this work is degraded, but it then recovers

I think of the implementation as pausing and resuming the reconciliation control loop(s). So relevant operators might go degraded, but only for reasons they go degraded prior to the introduction of change management.

Reply:
Behavior for upgrades today is we will cancel if it takes too long to start.  Would need to find out what "too long" actually is.  We'll need to be careful about how this is communicated when we switch over to using this for managed.  More thinking about the duration for control plane upgrades.  I know we don't guarentee how long it will take, but perhaps something should consider how long those upgrades take and estimate if it would roughly fit into the "window" of the schedule.

Reply:
Control plane averages about an hour. Are you thinking that OCM might want to prevent users from scheduling less than say, 2 hours to prevent their expectations from being violated?

Reply:
For control plane I wouldn't set an expectation on an end time, just allow scheduling the start.  Having an end implies an ability to stop the upgrade.  This is not the intention of this EP.